tests/test_parser.py

```diff
@@ -0,0 +1,58 @@
+import types
+
+import pytest
+
+from nexus.parser import Nexus, TokenType
+
+
+@pytest.mark.parametrize(
+    'nex,expect,roundtrip',
+    [
+        (
+            """#NEXUS
+BEGIN TREES;
+TREE best = (fish , (frog,
+(snake, mouse))) ;
+END;""",
+            lambda n: sum(1 for t in n.nexus if t.type != TokenType.WHITESPACE) == 23,
+            True),
+        (
+            '#NEXUS begin trees; tree ; end;',
+            lambda n: n.blocks[0].name == 'TREES',
+            True),
+        (
+            "#NEXUS BEGIN AssuMP[co[mm]ent]TiONS; END;",
+            lambda n: n.blocks[0].name == 'ASSUMPTIONS' and n.comments[0].text == 'co[mm]ent',
```

Comment:
Another example from the paper: Comments in words are allowed! But - at least in my implementation - prevent proper round-tripping, because I don't store in-word-position.

Reply:
hmm. comments in words, probably also means these sorts of things too:

https://github.com/phlorest/birchall_et_al2016/blob/095a592c9258a361de9f0ef738b18d7c8f4fd071/raw/relaxed-binary-simple.time.mcct.trees#L32

(i.e. substring from `[&length_range=...`) which is quite common. How hard is this to implement?